### PR TITLE
move checkDeps ask feature to fixDeps, add stopIfLoaded

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
This PR must not be merged before remindmodel is adapted to not use `checkDeps(action = "ask")` anymore. Also, in remindmodel a `renv::snapshot(prompt = FALSE)` is needed after `fixDeps(ask = TRUE)`.

### CODE REVIEW

**Comment by mikapfl** [comment]
Hm, I'm not a fan of changing API anymore. Moving things into piamenv already introduces essentially <= constraints on lucode2 versions in remindmodel, but we can't retroactively declare them. That's bad. I'd like to avoid breaking backwards compatibility if possible.

**Comment by pfuehrlich-pik** [comment]
You are right, thanks for criticizing! It should be possible to keep this backwards compatible quite easily now that I think about it. `fixDeps(ask= TRUE)` and `checkDeps(action = "ask")` will basically do the same thing, but we can probably live with that.

**Comment by pfuehrlich-pik** [comment]
This PR will no longer break the API, `checkDeps(action = "ask")` will continue to work, so this can safely be merged and will not break remind.

**Review by mikapfl — APPROVED** [review]

File: R/fixDeps.R [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if (!tolower(getLine()) %in% c("", "y", "yes")) {
+      return(invisible(list()))
+    }
+  }
+  unfixableReqs <- grep(":", unmetReqs, value = TRUE, invert = TRUE)
+
+  # the following sub(...) will transform strings like
+  # lucode2 == 1.2.3 is required, but 1.0.0 is installed - please install: lucode2@1.2.3
+  # into just "lucode2@1.2.3" which can be passed to renv::install
+  renvInstall <- sub("^.+: ([^:]+)$", "\\1", setdiff(unmetReqs, unfixableReqs))
```
Comment (mikapfl) [inline R/fixDeps.R:31]:
```suggestion
  ^:]
```
Look at happy Elvis! He's so happy!

(Not actually a suggested change, but the easiest way to highlight something in here)

Comment (pfuehrlich-pik) [reply R/fixDeps.R:31]: :joy:

**Pull request merged**